pkg/dogeboxd.go

```diff
@@ -276,6 +279,23 @@ func (t *Dogeboxd) updatePupProviders(j Job, u UpdatePupProviders) {
 		t.sendFinishedJob("action", j)
 		return
 	}
+
+	// Once we've updated our providers, we might need to rebuild
+	// some of our container configurations to fix up firewall rules.
```

Comment:
would be good to check here if there are exposes rules that would have changed and only rebuild if that were the case?

Reply:
I thought about this and it's kind of a rabbit hole. We'd need to store previously updated nix state somewhere and do comparisons. I think eventually we might get there, but for now we leave. Both the update + rebuild should effectively be noops and fairly fast if nothing has changed.